prow/pod-utils/gcs/upload.go

```diff
@@ -36,11 +40,34 @@ type UploadFunc func(writer dataWriter) error
 type destToWriter func(dest string) dataWriter
 
 // Upload uploads all of the data in the
-// uploadTargets map to GCS in parallel. The map is
-// keyed on GCS path under the bucket
-func Upload(bucket *storage.BucketHandle, uploadTargets map[string]UploadFunc) error {
+// uploadTargets map to blob storage in parallel. The map is
+// keyed on blob storage path under the bucket
+func Upload(bucket, gcsCredentialsFile, s3CredentialsFile string, uploadTargets map[string]UploadFunc) error {
+	parsedBucket, err := url.Parse(bucket)
+	if err != nil {
+		return fmt.Errorf("cannot parse bucket name %s: %v", bucket, err)
+	}
+	if parsedBucket.Scheme == "" || parsedBucket.Scheme == "gs" {
+		if parsedBucket.Scheme == "gs" {
+			bucket = strings.Trim(bucket, "gs://")
+		}
+		gcsClient, err := storage.NewClient(context.Background(), option.WithCredentialsFile(gcsCredentialsFile))
```

Comment:
Fine for me. I just preserved the previous behavior of `prow/gcsupload`. I think it will be less redundant once we replaced everything with the opener (as you said after we proved that s3 support works everywhere)